x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/RequestExecutorServiceSettings.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.http.sender;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class RequestExecutorServiceSettings {
+
+    /**
+     * The capacity of the internal queue. Zero is considered unlimited. If a positive value is used, the queue will reject entries
+     * once it is full.
+     */
+    static final Setting<Integer> TASK_QUEUE_CAPACITY_SETTING = Setting.intSetting(
+        "xpack.inference.http.request_executor.queue_capacity",
+        2000,
```

Comment:
I tried a reindex with a size of 3000 and max_docs 3000 and it worked fine. If we run into problems hopefully users can just adjust the size without us needing to add a code change.